CRM/Contribute/BAO/Contribution.php

```diff
@@ -5559,6 +5502,72 @@ protected static function getContributionPaymentLinks($id, $balance, $contributi
     return $actionLinks;
   }
 
+  /**
+   * @param array $contactIDs
+   *
+   * @return string
+   */
+  public static function getAnnualQuery($contactIDs) {
+    $contactIDs = implode(',', $contactIDs);
+    $config = CRM_Core_Config::singleton();
+    $currentMonth = date('m');
+    $currentDay = date('d');
+    if ((int ) $config->fiscalYearStart['M'] > $currentMonth ||
+      ((int ) $config->fiscalYearStart['M'] == $currentMonth &&
+        (int ) $config->fiscalYearStart['d'] > $currentDay
+      )
+    ) {
+      $year = date('Y') - 1;
+    }
+    else {
+      $year = date('Y');
+    }
+    $nextYear = $year + 1;
+
+    if ($config->fiscalYearStart) {
+      $newFiscalYearStart = $config->fiscalYearStart;
+      if ($newFiscalYearStart['M'] < 10) {
+        $newFiscalYearStart['M'] = '0' . $newFiscalYearStart['M'];
+      }
+      if ($newFiscalYearStart['d'] < 10) {
+        $newFiscalYearStart['d'] = '0' . $newFiscalYearStart['d'];
```

Comment:
Presumably this is to convert 1 into 01. I'm 50/50 on whether or a comment should say that's what this is doing.

Reply:
comment / todo added